### PR TITLE
Backport some memory improvements

### PR DESCRIPTION
**Pull request by will89**
* Adds this project to circleci
* Adds salsify_rubocop with everything ignored for now
* Enabled frozen string literal with included fixes for that `+''`
* Fix a couple of ruby 3 issues
* Removed some un-necessary array allocations

prime: @broels 
cc: @salsify/laser-viper

### CODE REVIEW

File: test/test_roo.rb [hunk 1]
```diff
@@ -1,1967 +1,1975 @@
-# encoding: utf-8
```
Comment (will89) [inline test/test_roo.rb:1]: Not sure why this diff is so crazy. I only added `# frozen_string_literal: true` and added some TODOs with Salsify among a few assertions that have probably failed on the mini tests for some time.

File: lib/roo/excelx.rb [hunk 2]
```diff
@@ -103,7 +105,8 @@ class Coordinate
       attr_accessor :row, :column
 
       def initialize(row, column)
-        @row, @column = row, column
```
Comment (will89) [inline lib/roo/excelx.rb:106]: This style actually generated an array

**Comment by will89** [comment]
For an excel file with 8695 rows (286KB) there was a 17% reduction in memory usage

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
+version: 2.1
+jobs:
+  lint:
+      docker:
+        - image: salsify/ruby_ci:2.6.6
```
Comment (broels) [inline .circleci/config.yml]: any reason not to fast-forward to `2.6.8` like with the test job?

File: lib/roo/excelx.rb (hunk 2 above)
Comment (broels) [reply lib/roo/excelx.rb:106]: I much prefer the updated form anyways, but curious is this a matter of ruby version? 
```
dandelion(dev)(writable):001:0> a, b = 1, 2
=> [1, 2]
dandelion(dev)(writable):002:0> a
=> 1
dandelion(dev)(writable):003:0> b
=> 2
```

File: test/test_generic_spreadsheet.rb [hunk 4]
```diff
@@ -1,4 +1,5 @@
 # -*- encoding: utf-8 -*-
+# frozen_string_literal: true
```
Comment (broels) [inline test/test_generic_spreadsheet.rb:2]: Couldn't help but notice there are a few different forms of this encoding declaration, any difference between all of these? Maybe it doesn't matter
```
# encoding: utf-8
```
```
# coding: utf-8
```
```
 # -*- encoding: utf-8 -*-
```

File: test/test_roo.rb [hunk 5]
```diff
@@ -1,1967 +1,1975 @@
...
+      assert_equal 6, oo.cell(2,2)
+      assert_equal 7, oo.cell(2,3)
+      assert_equal 8, oo.cell(2,4)
+      assert_equal 9, oo.cell(2,5)
+      assert_equal "test", oo.cell(2,6)
+      assert_equal :string, oo.celltype(2,6)
+      assert_equal 11, oo.cell(2,7)
+      # TODO: This has been failing for Salsify for a long time
+      # unless oo.kind_of? Roo::CSV
+      #   assert_equal :float, oo.celltype(2,7)
```
Comment (broels) [inline test/test_roo.rb:116]: Interesting, what are all of these if not floats? Of course I can't find any tooling in Excel that would tell us what to expect

**Review by broels — APPROVED** [review]
Only some questions for myself really, nice imp-roo-vement :shipit:

File: .circleci/config.yml (hunk 3 above)
Comment (will89) [reply .circleci/config.yml]: Good catch

File: test/test_generic_spreadsheet.rb (hunk 4 above)
Comment (will89) [reply test/test_generic_spreadsheet.rb:2]: Good question, I'm unsure

File: test/test_roo.rb (hunk 5 above)
Comment (will89) [reply test/test_roo.rb:116]: They're sometimes a `:string`. I don't think there's a type `:integer`. Float had bad side effects for integers in spreadsheets.

**Comment by will89** [comment]
I think the `a, b = 1, 2` is consistent across ruby versions as far as I know

**Pull request merged**